### ISSUE TITLE
Forbid `g` flag used on `RegExp#test()`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1174]
Got the idea from https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1173

If the regex is only used once on `RegExp#test()`, the `g` flag is useless (~maybe also forbid `y` flag?~)

#### Fail

```js
/foo/g.test(bar);
```

```js
new RegExp('foo', 'g').test(bar);
```

#### Pass

```js
const regex = /foo/g;
regex.test(bar);
```

I didn't mark this issue as rule proposal, because I was thinking maybe we can add this to [`better-regex`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/better-regex.md)?

**Comment by sindresorhus** [comment]
> (maybe also forbid y flag?)

👍

**Comment by fisker** [comment]
I was wrong about `y` flag 

```text
/foo/.test('-foo');
true
/foo/y.test('-foo');
false
```

**Comment by fisker** [comment]
Another possible problem in regex used to test. It should not capture group.

#### Fail

```js
/(foo)/.test('foo')
```

```js
/(?<named>foo)/.test('foo')
```

#### Pass

```js
/([-_])foo\1/.test('_foo-')
```

```js
/(?<separator>[-_])foo\k<separator>/.test('_foo-')
```

**Comment by sindresorhus** [comment]
@fisker Should we close this in favor of the regex ESLint plugin?

**Comment by fisker** [comment]
I've adopt `eslint-plugin-regex`, up to you.
